src/dynamics/tests/initial_conditions/ic_baroclinic.F90

```diff
@@ -306,12 +315,14 @@ subroutine bc_wav_set_ic(vcoord,latvals, lonvals, zint, U, V, T, PS, PHIS, &
                 pk = moist_pressure_given_z(zk(k),latvals(i))
                 qk = qv_given_moist_pressure(pk,latvals(i))
               else
-                qk = 0.d0
+                qk = 0._r8
+              end if
+              if (lq) then
+                Q(i,k,m_cnst_ix_q) = qk
               end if
-              if (lq .and. cnst1_is_moisture) Q(i,k,1) = qk
```

Comment:
Is the problem that Q was not getting set by ```bc_wav_set_ic```? While this is a clunkier way to set Q, shouldn't it still work? Would you mind explaining to me how this buggy code is resulting in Q incorrectly being zero everywhere?